src/main/java/org/wildfly/security/http/HttpAuthenticator.java

```diff
@@ -255,14 +255,20 @@ private boolean authenticate() throws HttpAuthenticationException {
                     statusCodeAllowed = true;
                     if (responders.size() > 0) {
                         boolean atLeastOneChallenge = false;
+
+                        int defaultStatusCode = OK;
                         boolean statusSet = false;
                         for (HttpServerMechanismsResponder responder : responders) {
                             try {
                                 responder.sendResponse(this);
                                 atLeastOneChallenge = true;
-                                if (statusSet == false && statusCode > 0 && statusCode != OK) {
-                                    httpExchangeSpi.setStatusCode(statusCode);
-                                    statusSet = true;
+                                if (!statusSet && statusCode > 0) {
+                                    if (statusCode == OK || statusCode == FORBIDDEN) { // minor status code change default
+                                        defaultStatusCode = statusCode;
```

Comment:
I think probably could use a minor tweak here.  I think once defaultStatusCode has been set to 403 it should not be possible to set it back to 200.  

With our current mechanism implementations that is not likely to happen, but with custom mechanisms you could risk flipping back and forth between 200 and 403.